iTechArt.Repositories/Repositories/Repository.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Data.Entity;
+using iTechArt.Repositories.Interfaces;
+using System.Data.Entity.Migrations;
+
+namespace iTechArt.Repositories.Repositories
+{
+    class Repository<T> : IRepository<T> where T : class
+    {
+        private readonly DbContext _dbContext;
+        private readonly DbSet<T> _dbSet;
+
+
+        public Repository(DbContext dbContext) 
+        {
+            _dbContext = dbContext;
+            _dbSet = _dbContext.Set<T>();
+        }
+
+
+        public IEnumerable<T> GetAll() 
+        {
+            return _dbSet;
+        }
+
+        public T GetById(int id)
+        {
+            return _dbSet.Find(id);
+        }
+
+        public void Add(T entity)
+        {
+            _dbSet.Add(entity);
+        }
+
+        public void Remove(T entity)
+        {
+            _dbSet.Remove(entity);
+        }
+
+        public void Update(T entity)
+        {
+            _dbSet.AddOrUpdate(entity);
+        }
+    }
```

Comment:
1. Investigate includes
2. Provide method for derived repositories to configure their includes